pkg/storage/cache/cache.go

```diff
@@ -0,0 +1,122 @@
+package cache
+
+import (
+	"context"
+	"github.com/logrange/linker"
+	"github.com/solarisdb/solaris/api/gen/solaris/v1"
+	"github.com/solarisdb/solaris/golibs/container/lru"
+	"github.com/solarisdb/solaris/pkg/storage"
+	"github.com/solarisdb/solaris/pkg/storage/logfs"
+)
+
+type (
+	// LogsChunksMetaStorage combines storage.Logs and
+	// logfs.LogsMetaStorage interfaces
+	LogsChunksMetaStorage interface {
+		logfs.LogsMetaStorage
+		storage.Logs
+	}
+
+	// CachedStorage wraps LogsChunksMetaStorage
+	// with caches for logs and chunks
+	CachedStorage struct {
+		storage        LogsChunksMetaStorage
+		logsCache      *lru.Cache[string, *solaris.Log]
+		chunksCache    *lru.Cache[string, []logfs.ChunkInfo]
+		lastChunkCache *lru.Cache[string, logfs.ChunkInfo]
```

Comment:
maybe use chunksCache for the  GetLastChunk(() as well? and the lastChunkCache will be not needed?

Reply:
applied